src/v/cluster/cloud_metadata/cluster_manifest.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "cloud_storage/base_manifest.h"
+#include "cluster/cloud_metadata/types.h"
+#include "model/fundamental.h"
+#include "seastarx.h"
+#include "serde/envelope.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/lowres_clock.hh>
+
+#include <rapidjson/document.h>
+
+namespace cluster::cloud_metadata {
+
+// Manifest containing information about cluster metadata.
+//
+// NOTE: while it's not a requirement to be serialized with serde, we'll use
+// serde-style versioning to simplify reasoning about compatibility.
+struct cluster_metadata_manifest
```

Comment:
Let's add a field that represents a schema version in the cloud. It could act as a guard rail for read replica in case if the source cluster uses newer version of redpanda and uploads some artefacts that read-replica can't process.

Reply:
This could also help with schema evolution.

Reply:
Just clarifying, by read-replicas, you mean recovery clusters right? Doesn't this version field act as a sort of schema version? At least for serde, the version field acts as a schema version, doesn't it? IIUC the `version` is updated on every new field, and `compat_version` is the lowest version we can serialize?

I agree having a schema version generally makes sense. I can have this be a separate field if there's concern over using serde version

Reply:
We chatted about this a bit. This would be more of a general-purpose schema used to differentiate potential format/naming changes to the underlying partition manifests. We agreed that there isn't necessarily a requirement for this yet since if/when we do need something like that, since this is JSON and serde, it's easy to extend with a new field.